python/cudf/dataframe/column.py

```diff
@@ -17,10 +18,19 @@
 
 from cudf.utils import cudautils, utils, ioutils
 from cudf.dataframe.buffer import Buffer
+from cudf.dataframe import columnops
 from cudf.bindings.cudf_cpp import count_nonzero_mask, column_view_pointer
 from cudf.bindings.concat import _column_concat
 
 
+@cuda.jit
+def boolean_assign_kernel(data, mask, out):
+    idx = cuda.threadIdx.x + cuda.threadIdx.y * cuda.blockDim.x
+    # assume data.size == mask.size == out.size
```

Comment:
Unsure if you're just working out the logic, but `data.size` will not equal `mask.size`. Mask is a bitarray, but you can't operate at the bit level, so the size is reported based on the logical dtype which will be at a minimum 1 byte.

Reply:
Unless this is a full on byte boolean mask and I'm dumb...

Reply:
That was written early in development but that mask refers to the array mask. So you might have
```python
data = [-1, -1, -1, -1]
mask = [0, 0, 1, 1]  # or equivalently [False, False, True, True]
out = [0, 1, 2, 3]
```
and `out` would end up as `[-1, -1, 2, 3]`